jgem fails to load gem from rubygems on Raspbian

### Environment
Hardware raspberrypi 3B+
* JRuby version (`jruby -v`)
jruby 9.2.0.0 (2.5.0) 2018-05-24 81156a8 Java HotSpot(TM) Client VM 25.65-b01 on 1.8.0_65-b17 +jit [linux-arm] (got same issue with jruby 9.1.17.0, and possibly earlier versions also fail)
* Operating system and platform (e.g. `uname -a`)
Linux raspberrypi 4.14.42-v7+ #1114 SMP Mon May 21 16:39:21 BST 2018 armv7l GNU/Linux
Other relevant info you may wish to add:
* GEM_HOME
/home/tux/.gem/ruby/2.3.0
*ruby version
ruby 2.3.3p222 (2016-11-21) [arm-linux-gnueabihf]
* java version
java version "1.8.0_65"
Java(TM) SE Runtime Environment (build 1.8.0_65-b17)
Java HotSpot(TM) Client VM (build 25.65-b01, mixed mode)

mri irb (jirb also failing)
puts RbConfig::CONFIG['host_os']
`linux-gnueabihf`

This might be significant with jruby, `RbConfig::CONFIG['host_os']` returns `linux` ie close to useless.


### Expected Behavior
`gem install toxiclibs`
mri ruby installs as expected

### Actual Behavior
jgem install toxiclibs
Fetching: toxiclibs-1.0.0.gem (100%)
NotImplementedError: flock unsupported or native support failed to load; see http://wiki.jruby.org/Native-Libraries
                   flock at org/jruby/RubyFile.java:322
   block in write_binary at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems.rb:897
                    open at org/jruby/RubyIO.java:1179
                    open at org/jruby/RubyKernel.java:311
            write_binary at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems.rb:895
       cache_update_path at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/remote_fetcher.rb:334
                download at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/remote_fetcher.rb:188
                download at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/source.rb:215
                 install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/resolver/specification.rb:91
        block in install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/request_set.rb:168
                    each at org/jruby/RubyArray.java:1801
                 install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/request_set.rb:156
             install_gem at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:205
   block in install_gems at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:255
                    each at org/jruby/RubyArray.java:1801
            install_gems at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:251
                 execute at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:158
  invoke_with_build_args at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command.rb:313
            process_args at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command_manager.rb:171
                     run at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command_manager.rb:141
                     run at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/gem_runner.rb:59
                  <main> at /usr/bin/jgem:21
### Report
The error message is opaque to me, I'm a seasoned linux user, but very new to raspbian OS, from searches I understood that in the past same error had been reported (different OS etc) but it was unclear to me whether it had actually been resolved. I seem to remember seeing something about failure in bit detection. Raspbian is odd in being 32 bit on 64 bit processor (these days) but probably for good reason. I have been developing [PiCrate](https://github.com/ruby-processing/PiCrate) on my 64 bit linux mint machine and never had this problem before (similar setup). PS: as a workaround to install the picrate gem, I used a local install with jgem without issue.

### Another gem

jruby -Xnative.verbose=true -S gem install geomerative
Failed to load native POSIX impl; falling back on Java impl. Stacktrace follows.
java.lang.UnsatisfiedLinkError: could not load FFI provider jnr.ffi.provider.jffi.Provider
	at jnr.ffi.provider.InvalidProvider$1.loadLibrary(InvalidProvider.java:48)
	at jnr.ffi.LibraryLoader.load(LibraryLoader.java:325)
	at jnr.ffi.Library.loadLibrary(Library.java:127)
	at jnr.posix.POSIXFactory$DefaultLibCProvider$SingletonHolder.<clinit>(POSIXFactory.java:279)
	at jnr.posix.POSIXFactory$DefaultLibCProvider.getLibC(POSIXFactory.java:283)
	at jnr.posix.BaseNativePOSIX.<init>(BaseNativePOSIX.java:37)
	at jnr.posix.LinuxPOSIX.<init>(LinuxPOSIX.java:19)
	at jnr.posix.POSIXFactory.loadLinuxPOSIX(POSIXFactory.java:143)
	at jnr.posix.POSIXFactory.loadNativePOSIX(POSIXFactory.java:121)
	at jnr.posix.POSIXFactory.loadPOSIX(POSIXFactory.java:90)
	at jnr.posix.LazyPOSIX.loadPOSIX(LazyPOSIX.java:38)
	at jnr.posix.LazyPOSIX.posix(LazyPOSIX.java:32)
	at jnr.posix.LazyPOSIX.isNative(LazyPOSIX.java:398)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:287)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1238)
	at org.jruby.Ruby.newInstance(Ruby.java:362)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
Caused by: java.lang.UnsatisfiedLinkError: could not get native definition for type: POINTER
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:251)
	at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
	at com.kenai.jffi.Type.resolveSize(Type.java:155)
	at com.kenai.jffi.Type.size(Type.java:138)
	at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
	at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
	at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
	at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
	at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
	at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
	at jnr.ffi.Library.loadLibrary(Library.java:114)
	... 17 more
Caused by: java.lang.UnsatisfiedLinkError: java.lang.UnsatisfiedLinkError: /tmp/jffi8383875517230482219.so: /tmp/jffi8383875517230482219.so: cannot open shared object file: No such file or directory
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1938)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1821)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at com.kenai.jffi.internal.StubLoader.loadFromJar(StubLoader.java:376)
	at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:258)
	at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:449)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.kenai.jffi.Init.load(Init.java:68)
	at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
	at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
	at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
	at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
	at com.kenai.jffi.Type.resolveSize(Type.java:155)
	at com.kenai.jffi.Type.size(Type.java:138)
	at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
	at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
	at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
	at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
	at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
	at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
	at jnr.ffi.Library.loadLibrary(Library.java:114)
	at jnr.posix.POSIXFactory$DefaultLibCProvider$SingletonHolder.<clinit>(POSIXFactory.java:279)
	at jnr.posix.POSIXFactory$DefaultLibCProvider.getLibC(POSIXFactory.java:283)
	at jnr.posix.BaseNativePOSIX.<init>(BaseNativePOSIX.java:37)
	at jnr.posix.LinuxPOSIX.<init>(LinuxPOSIX.java:19)
	at jnr.posix.POSIXFactory.loadLinuxPOSIX(POSIXFactory.java:143)
	at jnr.posix.POSIXFactory.loadNativePOSIX(POSIXFactory.java:121)
	at jnr.posix.POSIXFactory.loadPOSIX(POSIXFactory.java:90)
	at jnr.posix.LazyPOSIX.loadPOSIX(LazyPOSIX.java:38)
	at jnr.posix.LazyPOSIX.posix(LazyPOSIX.java:32)
	at jnr.posix.LazyPOSIX.isNative(LazyPOSIX.java:398)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:287)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1238)
	at org.jruby.Ruby.newInstance(Ruby.java:362)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
java.lang.UnsatisfiedLinkError: /opt/jruby-9.2.0.0/lib/jni/arm-Linux/libjffi-1.2.so: /opt/jruby-9.2.0.0/lib/jni/arm-Linux/libjffi-1.2.so: cannot open shared object file: No such file or directory
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1938)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1821)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at com.kenai.jffi.internal.StubLoader.loadFromBootPath(StubLoader.java:321)
	at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:248)
	at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:449)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.kenai.jffi.Init.load(Init.java:68)
	at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
	at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
	at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
	at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
	at com.kenai.jffi.Type.resolveSize(Type.java:155)
	at com.kenai.jffi.Type.size(Type.java:138)
	at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
	at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
	at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
	at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
	at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
	at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
	at jnr.ffi.Library.loadLibrary(Library.java:114)
	at jnr.posix.POSIXFactory$DefaultLibCProvider$SingletonHolder.<clinit>(POSIXFactory.java:279)
	at jnr.posix.POSIXFactory$DefaultLibCProvider.getLibC(POSIXFactory.java:283)
	at jnr.posix.BaseNativePOSIX.<init>(BaseNativePOSIX.java:37)
	at jnr.posix.LinuxPOSIX.<init>(LinuxPOSIX.java:19)
	at jnr.posix.POSIXFactory.loadLinuxPOSIX(POSIXFactory.java:143)
	at jnr.posix.POSIXFactory.loadNativePOSIX(POSIXFactory.java:121)
	at jnr.posix.POSIXFactory.loadPOSIX(POSIXFactory.java:90)
	at jnr.posix.LazyPOSIX.loadPOSIX(LazyPOSIX.java:38)
	at jnr.posix.LazyPOSIX.posix(LazyPOSIX.java:32)
	at jnr.posix.LazyPOSIX.isNative(LazyPOSIX.java:398)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:287)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1238)
	at org.jruby.Ruby.newInstance(Ruby.java:362)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)

	at com.kenai.jffi.Foreign.newLoadError(Foreign.java:72)
	at com.kenai.jffi.Foreign.access$300(Foreign.java:42)
	at com.kenai.jffi.Foreign$InValidInstanceHolder.getForeign(Foreign.java:98)
	at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
	... 36 more
Caused by: java.lang.UnsatisfiedLinkError: java.lang.UnsatisfiedLinkError: /tmp/jffi8383875517230482219.so: /tmp/jffi8383875517230482219.so: cannot open shared object file: No such file or directory
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1938)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1821)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at com.kenai.jffi.internal.StubLoader.loadFromJar(StubLoader.java:376)
	at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:258)
	at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:449)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.kenai.jffi.Init.load(Init.java:68)
	at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
	at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
	at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
	at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
	at com.kenai.jffi.Type.resolveSize(Type.java:155)
	at com.kenai.jffi.Type.size(Type.java:138)
	at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
	at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
	at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
	at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
	at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
	at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
	at jnr.ffi.Library.loadLibrary(Library.java:114)
	at jnr.posix.POSIXFactory$DefaultLibCProvider$SingletonHolder.<clinit>(POSIXFactory.java:279)
	at jnr.posix.POSIXFactory$DefaultLibCProvider.getLibC(POSIXFactory.java:283)
	at jnr.posix.BaseNativePOSIX.<init>(BaseNativePOSIX.java:37)
	at jnr.posix.LinuxPOSIX.<init>(LinuxPOSIX.java:19)
	at jnr.posix.POSIXFactory.loadLinuxPOSIX(POSIXFactory.java:143)
	at jnr.posix.POSIXFactory.loadNativePOSIX(POSIXFactory.java:121)
	at jnr.posix.POSIXFactory.loadPOSIX(POSIXFactory.java:90)
	at jnr.posix.LazyPOSIX.loadPOSIX(LazyPOSIX.java:38)
	at jnr.posix.LazyPOSIX.posix(LazyPOSIX.java:32)
	at jnr.posix.LazyPOSIX.isNative(LazyPOSIX.java:398)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:287)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1238)
	at org.jruby.Ruby.newInstance(Ruby.java:362)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
java.lang.UnsatisfiedLinkError: /opt/jruby-9.2.0.0/lib/jni/arm-Linux/libjffi-1.2.so: /opt/jruby-9.2.0.0/lib/jni/arm-Linux/libjffi-1.2.so: cannot open shared object file: No such file or directory
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1938)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1821)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at com.kenai.jffi.internal.StubLoader.loadFromBootPath(StubLoader.java:321)
	at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:248)
	at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:449)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.kenai.jffi.Init.load(Init.java:68)
	at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
	at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
	at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
	at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
	at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
	at com.kenai.jffi.Type.resolveSize(Type.java:155)
	at com.kenai.jffi.Type.size(Type.java:138)
	at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
	at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
	at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
	at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
	at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
	at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
	at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
	at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
	at jnr.ffi.Library.loadLibrary(Library.java:114)
	at jnr.posix.POSIXFactory$DefaultLibCProvider$SingletonHolder.<clinit>(POSIXFactory.java:279)
	at jnr.posix.POSIXFactory$DefaultLibCProvider.getLibC(POSIXFactory.java:283)
	at jnr.posix.BaseNativePOSIX.<init>(BaseNativePOSIX.java:37)
	at jnr.posix.LinuxPOSIX.<init>(LinuxPOSIX.java:19)
	at jnr.posix.POSIXFactory.loadLinuxPOSIX(POSIXFactory.java:143)
	at jnr.posix.POSIXFactory.loadNativePOSIX(POSIXFactory.java:121)
	at jnr.posix.POSIXFactory.loadPOSIX(POSIXFactory.java:90)
	at jnr.posix.LazyPOSIX.loadPOSIX(LazyPOSIX.java:38)
	at jnr.posix.LazyPOSIX.posix(LazyPOSIX.java:32)
	at jnr.posix.LazyPOSIX.isNative(LazyPOSIX.java:398)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:287)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1238)
	at org.jruby.Ruby.newInstance(Ruby.java:362)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)

	at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:270)
	at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:449)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.kenai.jffi.Init.load(Init.java:68)
	at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
	at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
	... 38 more
Fetching: geomerative-1.0.2-java.gem (100%)
NotImplementedError: flock unsupported or native support failed to load; see http://wiki.jruby.org/Native-Libraries
                   flock at org/jruby/RubyFile.java:322
   block in write_binary at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems.rb:897
                    open at org/jruby/RubyIO.java:1179
                    open at org/jruby/RubyKernel.java:311
            write_binary at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems.rb:895
       cache_update_path at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/remote_fetcher.rb:334
                download at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/remote_fetcher.rb:188
                download at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/source.rb:215
                 install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/resolver/specification.rb:91
        block in install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/request_set.rb:168
                    each at org/jruby/RubyArray.java:1801
                 install at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/request_set.rb:156
             install_gem at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:205
   block in install_gems at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:255
                    each at org/jruby/RubyArray.java:1801
            install_gems at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:251
                 execute at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/commands/install_command.rb:158
  invoke_with_build_args at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command.rb:313
            process_args at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command_manager.rb:171
                     run at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/command_manager.rb:141
                     run at /opt/jruby-9.2.0.0/lib/ruby/stdlib/rubygems/gem_runner.rb:59
                  <main> at /opt/jruby-9.2.0.0/bin/jgem:21
                    load at org/jruby/RubyKernel.java:994
                  <main> at /opt/jruby-9.2.0.0/bin/gem:4

--- a/pom.rb
+++ b/pom.rb
@@ -83,7 +83,7 @@ project 'JRuby', 'https://github.com/jruby/jruby' do
               'jruby-launcher.version' => '1.1.6',
               'ant.version' => '1.9.8',
               'asm.version' => '6.2.1',
-              'jffi.version' => '1.2.17',
+              'jffi.version' => '1.2.18',
               'joda.time.version' => '2.9.9' )
 
   plugin_management do

--- a/pom.xml
+++ b/pom.xml
@@ -113,7 +113,7 @@ DO NOT MODIFIY - GENERATED CODE
     <its.j2ee>j2ee*/pom.xml</its.j2ee>
     <its.osgi>osgi*/pom.xml</its.osgi>
     <jar-dependencies.version>0.4.0</jar-dependencies.version>
-    <jffi.version>1.2.17</jffi.version>
+    <jffi.version>1.2.18</jffi.version>
     <joda.time.version>2.9.9</joda.time.version>
     <jruby-launcher.version>1.1.6</jruby-launcher.version>
     <jruby.basedir>${project.basedir}</jruby.basedir>

